### PR TITLE
Fix network status detection

### PR DESCRIPTION
**Pull request by antonmolodykh**
Refs #579

### CODE REVIEW

File: lib/utils.py [hunk 1]
```diff
@@ -57,14 +57,24 @@ def validate_url(string):
...
-        active_interfaces = interfaces()
 
-        if 'eth0' in active_interfaces:
-            interface = 'eth0'
-        elif 'wlan0' in active_interfaces:
-            interface = 'wlan0'
-        else:
-            raise Exception("No active network interface found.")
+        interface = None
+        for ifc in ['eth0', 'wlan0']:
```
Comment (vpetersson) [inline lib/utils.py]: I don't think this works on `wlan0`? It's a physical link, so I don't think this concept is applicable.

Comment (antonmolodykh) [inline lib/utils.py]: Tomorrow I will have a new PI and I'll check it out

File: lib/utils.py [hunk 2]
```diff
@@ -57,14 +57,24 @@ def validate_url(string):
...
-            interface = 'wlan0'
-        else:
-            raise Exception("No active network interface found.")
+        interface = None
+        try:
+            file_carrier = open('/sys/class/net/eth0/carrier')
+            file_operstate = open('/sys/class/net/eth0/operstate')
+
+            if "1" in file_carrier.read() and "up" in file_operstate.read():
+                interface = 'eth0'
```
Comment (vpetersson) [inline lib/utils.py]: Actually, take a look at this:
https://github.com/Screenly/screenly-ose/blob/master/ansible/roles/network/files/screenly_net_mgr.py#L157-L162

File: lib/utils.py [hunk 3]
```diff
@@ -57,14 +57,30 @@ def validate_url(string):
...
+        if not interface:
+            file_interfaces = open('/etc/network/interfaces')
+            for n in range(10):
+                iface = 'wlan{}'.format(n)
+                if iface in file_interfaces.read():
+                    interface = iface
+                    break
+
+        if not interface:
+            raise Exception("No active network connection found")
```
Comment (vpetersson) [inline lib/utils.py]: Add trailing period.

**Pull request merged**